system/updated/updated.py

```diff
@@ -0,0 +1,178 @@
+  except Exception:
+    cloudlog.exception("fetching remote channels")
+    return None
+
+
+def get_remote_channel_data(channel) -> tuple[BuildMetadata | None, dict | None]:
+  try:
+    data = requests.get(f"{API_HOST}/{CHANNELS_API_ROOT}/{channel}").json()
+    return build_metadata_from_dict(data["build_metadata"]), data["manifest"]
+  except Exception:
+    cloudlog.exception("fetching remote manifest failed")
+    return None, None
+
+
+def check_update_available(current_directory, other_metadata: BuildMetadata):
+  build_metadata = get_build_metadata(current_directory)
+  return build_metadata.channel != other_metadata.channel or \
+         build_metadata.openpilot.git_commit != other_metadata.openpilot.git_commit or \
+         build_metadata.openpilot.build_style != other_metadata.openpilot.build_style
+
+
+def extract_directory_helper(entry, cache_directory, directory):
+  target = casync.parse_caibx(entry["casync"]["caibx"])
+
+  cache_filename = os.path.join(CASYNC_TMPDIR, "cache.tar")
+  tmp_filename = os.path.join(CASYNC_TMPDIR, "tmp.tar")
+
+  cloudlog.info("building tarball update cache...")
+  start = time.monotonic()
+  sources = [('cache', casync.DirectoryTarChunkReader(cache_directory, cache_filename), casync.build_chunk_dict(target))]
+  cloudlog.info(f"tarball cache creation completed in {time.monotonic() - start} seconds")
+
+  sources += [('remote', casync.RemoteChunkReader(casync.get_default_store(entry["casync"]["caibx"])), casync.build_chunk_dict(target))]
+
+  cloudlog.info(f"extracting {entry['casync']['caibx']} to {directory}")
+  start = time.monotonic()
+  stats = casync.extract_directory(target, sources, directory, tmp_filename)
+  cloudlog.info(f"extraction completed in {time.monotonic() - start} seconds with {stats}")
+
+
+
+# TODO: this can be removed after all devices have moved away from overlay based git updater
+OVERLAY_MERGED = os.path.join(STAGING_ROOT, "merged")
+
+def dismount_overlay() -> None:
+  if os.path.ismount(OVERLAY_MERGED):
+    cloudlog.info("unmounting existing overlay")
+    run_cmd(["sudo", "umount", "-l", OVERLAY_MERGED])
+
+
+def setup_dirs():
+  dismount_overlay()
+  run_cmd(["sudo", "rm", "-rf", STAGING_ROOT])
+  if os.path.isdir(STAGING_ROOT):
+    shutil.rmtree(STAGING_ROOT)
+
+  Path(STAGING_ROOT).mkdir()
+  CASYNC_TMPDIR.mkdir()
+  CASYNC_STAGING.mkdir()
+
+
+def download_update(manifest):
+  cloudlog.info(f"downloading update from: {manifest}")
+
+  for entry in manifest:
+    if entry["type"] == "path_tarred" and entry["path"] == "/data/openpilot":
+      extract_directory_helper(entry, BASEDIR, CASYNC_STAGING)
+
+  if AGNOS:
+    target_slot_number = get_target_slot_number()
+    cloudlog.info(f"Target slot {target_slot_number}")
+
+    # set target slot as unbootable
+    os.system(f"abctl --set_unbootable {target_slot_number}")
+
+    for entry in manifest:
+      if entry["type"] == "partition":
+        flash_partition(target_slot_number, entry, cloudlog)
```

Comment:
so this is a lot better than before, but can we remove the "if AGNOS" and "if openpilot"?